### PR TITLE
WIP: Positional audio modules

### PR DESCRIPTION
**Pull request by kschzt**

### CODE REVIEW

File: browser/plugins/audio_spatial_listener.plugin.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+(function() {
+
+var SpatialListener = E2.plugins.audio_spatial_listener = function(core, node) 
+{
+	this.desc = 'Spatial Listener.';
+	
+	this.input_slots = [ 
+		{ name: 'position', dt: core.datatypes.VECTOR, desc: 'Position.', def: core.renderer.vector_origin },
+		{ name: 'forward', dt: core.datatypes.VECTOR, desc: 'Forward Vector.', def: core.renderer.vector_origin },
+		{ name: 'up', dt: core.datatypes.VECTOR, desc: 'Up Vector.', def: core.renderer.vector_origin }
```
Comment (kkananen) [inline browser/plugins/audio_spatial_listener.plugin.js]: Default should be 0,1,0 probably - avoids you having to override this each time.
Also, isn't up always 0,1,0?

File: browser/plugins/audio_spatial_listener.plugin.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
+(function() {
+
+var SpatialListener = E2.plugins.audio_spatial_listener = function(core, node) 
+{
+	this.desc = 'Spatial Listener.';
+	
+	this.input_slots = [ 
+		{ name: 'position', dt: core.datatypes.VECTOR, desc: 'Position.', def: core.renderer.vector_origin },
+		{ name: 'forward', dt: core.datatypes.VECTOR, desc: 'Forward Vector.', def: core.renderer.vector_origin },
```
Comment (kkananen) [inline browser/plugins/audio_spatial_listener.plugin.js]: Default shouldn't be zero. In reality, is this ever anything else than 0,0,-1? In which case do we need this input at all?

**Comment by lpn** [comment]
The correct Up vector needs to be calculated between VR cam and the listener object

**Comment by kschzt** [comment]
lgtm

**Pull request merged**